tls/test_tls_cert.py

```diff
@@ -448,21 +447,21 @@ def start_all(self):
 
     def check_handshake_success(self, sni):
         """Run TLS handshake with the given SNI and check it is completes successfully."""
-        hs = TlsHandshake(verbose=self.verbose)
-        hs.sni = [sni]
+        hs = TlsHandshake()
+        hs.sni = sni
         # TLS 1.2 handshake completed with no exception => SNI is accepted
-        hs._do_12_hs()
-        self.assertTrue(x509_check_cn(hs.cert, "tempesta-tech.com"))
+        hs.do_12()
```

Comment:
`_do_12_hs()` was used here to make handshake, without sending the data. It is not possible with the new inplementation?

Reply:
Its possible, but by default we prepare one TLSAppData packet to send. It can be overrided